deepcell/training/model_reconstitution_test.py

```diff
@@ -0,0 +1,198 @@
+        assert self.source_metadata["deepcell_version"] == deepcell.__version__
+
+    def _load_data(self):
+        # take in dataset
+        filename = 'HeLa_S3.npz'
+        test_size = 0.1 # % of data saved as test
+        (self.X_train, self.y_train), (self.X_test, self.y_test) = deepcell.datasets.hela_s3.load_data(
+                                                                   filename,
+                                                                   test_size = test_size,
+                                                                   seed = self.seed_value)
+
+    def _compare_data(self):
+        assert tuple(self.source_metadata["dataset"]["training"]["X_train_shape"]) == self.X_train.shape
+        assert self.source_metadata["dataset"]["training"]["X_train_datatype"] == self.X_train.dtype.name
+        assert self.source_metadata["dataset"]["training"]["X_train_md5_digest"] == hashlib.md5(self.X_train).hexdigest()
+        assert tuple(self.source_metadata["dataset"]["training"]["y_train_shape"]) == self.y_train.shape
+        assert self.source_metadata["dataset"]["training"]["y_train_datatype"] == self.y_train.dtype.name
+        assert self.source_metadata["dataset"]["training"]["y_train_md5_digest"] == hashlib.md5(self.y_train).hexdigest()
+        assert tuple(self.source_metadata["dataset"]["testing"]["X_test_shape"]) == self.X_test.shape
+        assert self.source_metadata["dataset"]["testing"]["X_test_datatype"] == self.X_test.dtype.name
+        assert self.source_metadata["dataset"]["testing"]["X_test_md5_digest"] == hashlib.md5(self.X_test).hexdigest()
+        assert tuple(self.source_metadata["dataset"]["testing"]["y_test_shape"]) == self.y_test.shape
+        assert self.source_metadata["dataset"]["testing"]["y_test_datatype"] == self.y_test.dtype.name
+        assert self.source_metadata["dataset"]["testing"]["y_test_md5_digest"] == hashlib.md5(self.y_test).hexdigest()
+
+    def _create_generators(self):
+        ## training image generator
+        self.train_generator = deepcell.ImageFullyConvDataGenerator()
+        ## validation image generator
+        self.validation_generator = deepcell.ImageFullyConvDataGenerator()
+        #train_iterator = deepcell.ImageFullyConvIterator(
+        #                    train_dict = train_dict,
+        #                    image_data_generator = train_generator)
+        #validation_iterator = deepcell.ImageFullyConvIterator(
+        #                    train_dict = validation_dict,
+        #                    image_data_generator = train_generator)
+
+        ## read in parameters from metadata
+        for generator_type in ("train_generator", "validation_generator"):
+            if self.source_metadata["generators"][generator_type]["parameters"]["seed"]:
+                self.generator_seed = self.source_metadata["generators"][generator_type]["parameters"]["seed"]
+            else:
+                self.generator_seed = 43
```

Comment:
maybe this should be a default_seed or something configurable instead of just `43`.